src/ousd/post-processors/daily-stats/daily-stats.ts

```diff
@@ -0,0 +1,245 @@
+    await Promise.all([
+      ctx.store.findOne(OUSDAPY, queryParams),
+      ctx.store.findOne(OUSD, queryParams),
+      ctx.store.findOne(OUSDVault, queryParams),
+      ctx.store.findOne(OUSDMorphoAave, queryParams),
+      ctx.store.findOne(OUSDHistory, {
+        where: {
+          timestamp: LessThanOrEqual(date),
+          address: { id: '0xd2af830e8cbdfed6cc11bab697bb25496ed6fa62' },
+        },
+        order: { timestamp: 'desc' as FindOptionsOrderValue },
+      }),
+    ])
+
+  // Do we have any useful data yet?
+  const allEntities = [lastApy, lastOusd, lastVault, lastMorpho].filter(Boolean)
+  if (![lastApy, lastOusd].every((entity) => !!entity)) {
+    return null
+  }
+
+  const entityManager = (
+    ctx.store as unknown as {
+      em: () => EntityManager
+    }
+  ).em()
+
+  const end = dayjs.utc(date).endOf('day').toDate()
+  const yieldStats = await entityManager.query<
+    {
+      period: string
+      total_yield: bigint
+      total_fees: bigint
+      total_revenue: bigint
+      total_yield_converted: bigint
+      total_fees_converted: bigint
+      total_revenue_converted: bigint
+    }[]
+  >(yieldStatsQuery, [end])
+
+  const mostRecentEntity = allEntities.reduce((highest, current) => {
+    if (!highest || !current) return current
+    return current.blockNumber > highest.blockNumber ? current : highest
+  })
+
+  if (!mostRecentEntity?.blockNumber) {
+    return null
+  }
+
+  const id = date.toISOString().substring(0, 10)
+
+  const dailyStat = new OUSDDailyStat({
+    id,
+    blockNumber: mostRecentEntity?.blockNumber,
+    timestamp: mostRecentEntity?.timestamp,
+
+    apr: lastApy?.apr,
+    apy: lastApy?.apy,
+    apy7DayAvg: lastApy?.apy7DayAvg,
+    apy14DayAvg: lastApy?.apy14DayAvg,
+    apy30DayAvg: lastApy?.apy30DayAvg,
+
+    totalSupply: lastOusd?.totalSupply || 0n,
+    totalSupplyUSD: 0,
+    rebasingSupply: lastOusd?.rebasingSupply || 0n,
+    nonRebasingSupply: lastOusd?.nonRebasingSupply || 0n,
+    amoSupply: 0n,
+    dripperWETH: 0n,
+    wrappedSupply: lastWrappedOUSDHistory?.balance || 0n,
+
+    yield: yieldStats[0].total_yield || 0n,
+    fees: yieldStats[0].total_fees || 0n,
+    revenue: yieldStats[0].total_revenue || 0n,
+    revenue7DayAvg: BigInt(yieldStats[1].total_revenue || 0n) / 7n,
+    revenue7DayTotal: yieldStats[1].total_revenue || 0n,
+    revenueAllTime: yieldStats[2].total_revenue || 0n,
+
+    yieldConverted: yieldStats[0].total_yield_converted || 0n,
+    feesConverted: yieldStats[0].total_fees_converted || 0n,
+    revenueConverted: yieldStats[0].total_revenue_converted || 0n,
+    revenue7DayAvgConverted: BigInt(yieldStats[1].total_revenue_converted || 0n) / 7n,
+    revenue7DayTotalConverted: yieldStats[1].total_revenue_converted || 0n,
+    revenueAllTimeConverted: yieldStats[2].total_revenue_converted || 0n,
+
+    pegPrice: 0n,
+  })
+
+  const dailyStrategyStats: OUSDStrategyDailyStat[] = [
+    // new OUSDStrategyDailyStat({
```

Comment:
do we need this commented code?

Reply:
It's there for convenience when we add the strategy stats. A reminder those still need to be added